v19.1 with SSR not building/responding in dev mode: The operation was aborted due to timeout

### Command

build

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

19.0

### Description

Hey. Updating v19.0.x app to v19.1.x breaks build with this error:


✘ [ERROR] An error occurred while extracting routes.

Routes extraction was aborted.
TimeoutError: The operation was aborted due to timeout

No routes, no code, no logic.
Minimal repro available here: https://github.com/zip-fa/19-1-endless-build

Switching from server config:

```json
"server": "src/main.server.ts",
        "outputMode": "server",
        "ssr": {
          "entry": "src/server.ts"
        }
```

To static config:

```json
"outputMode": "static"
```
immediately helps

### Minimal Reproduction

nx build 19-1-endless-build

### Exception or Error

```text

```

### Your Environment

```text
ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 19.1.2
Node: 22.13.0
Package Manager: npm 10.9.2
OS: darwin arm64

Angular: 19.1.1
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1901.2
@angular-devkit/build-angular   19.1.2
@angular-devkit/core            19.1.2
@angular-devkit/schematics      19.1.2
@angular/cli                    19.1.2
@angular/ssr                    19.1.2
@schematics/angular             19.1.2
rxjs                            7.8.1
typescript                      5.7.3
zone.js                         0.15.0
```

### Anything else relevant?

_No response_

--- a/packages/angular/ssr/src/routes/ng-routes.ts
+++ b/packages/angular/ssr/src/routes/ng-routes.ts
@@ -523,11 +523,17 @@ export async function getRoutesFromAngularRouterConfig(
       applicationRef = await bootstrap();
     }
 
+    const injector = applicationRef.injector;
+    const router = injector.get(Router);
+
+    // Workaround to unblock navigation when `withEnabledBlockingInitialNavigation()` is used.
+    // This is necessary because route extraction disables component bootstrapping.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (router as any).navigationTransitions.afterPreactivation()?.next?.();
+
     // Wait until the application is stable.
     await applicationRef.whenStable();
 
-    const injector = applicationRef.injector;
-    const router = injector.get(Router);
     const routesResults: RouteTreeNodeMetadata[] = [];
     const errors: string[] = [];
 

--- a/packages/angular/ssr/test/routes/ng-routes_spec.ts
+++ b/packages/angular/ssr/test/routes/ng-routes_spec.ts
@@ -12,6 +12,7 @@ import '@angular/compiler';
 /* eslint-enable import/no-unassigned-import */
 
 import { Component } from '@angular/core';
+import { Routes, provideRouter, withEnabledBlockingInitialNavigation } from '@angular/router';
 import { extractRoutesAndCreateRouteTree } from '../../src/routes/ng-routes';
 import { PrerenderFallback, RenderMode } from '../../src/routes/route-config';
 import { setAngularAppTestingManifest } from '../testing-utils';
@@ -489,6 +490,38 @@ describe('extractRoutesAndCreateRouteTree', () => {
       undefined,
       undefined,
       RootComponent,
+    );
+
+    const { routeTree, errors } = await extractRoutesAndCreateRouteTree({ url });
+    expect(errors).toHaveSize(0);
+    expect(routeTree.toObject()).toHaveSize(2);
+  });
+
+  it('should not bootstrap the root component when using `withEnabledBlockingInitialNavigation`', async () => {
+    @Component({
+      standalone: true,
+      selector: 'app-root',
+      template: '',
+    })
+    class RootComponent {
+      constructor() {
+        throw new Error('RootComponent should not be bootstrapped.');
+      }
+    }
+
+    const routes: Routes = [
+      { path: '', component: DummyComponent },
+      { path: 'home', component: DummyComponent },
+    ];
+
+    setAngularAppTestingManifest(
+      routes,
+      [{ path: '**', renderMode: RenderMode.Server }],
+      undefined,
+      undefined,
+      undefined,
+      RootComponent,
+      [provideRouter(routes, withEnabledBlockingInitialNavigation())],
     );
 
     const { routeTree, errors } = await extractRoutesAndCreateRouteTree({ url });

--- a/packages/angular/ssr/test/testing-utils.ts
+++ b/packages/angular/ssr/test/testing-utils.ts
@@ -6,7 +6,13 @@
  * found in the LICENSE file at https://angular.dev/license
  */
 
-import { Component, Type, provideExperimentalZonelessChangeDetection } from '@angular/core';
+import {
+  Component,
+  EnvironmentProviders,
+  Provider,
+  Type,
+  provideExperimentalZonelessChangeDetection,
+} from '@angular/core';
 import { bootstrapApplication } from '@angular/platform-browser';
 import { provideServerRendering } from '@angular/platform-server';
 import { RouterOutlet, Routes, provideRouter } from '@angular/router';
@@ -35,6 +41,8 @@ class AppComponent {}
  *                                  where the keys are asset paths and the values are asset details.
  * @param locale - An optional locale to configure for the application during testing.
  * @param rootComponent - The root Angular component to bootstrap the application.
+ * @param extraProviders - An optional array of additional providers that should be available to the
+ *                         root component and all its children.
  */
 export function setAngularAppTestingManifest(
   routes: Routes,
@@ -43,6 +51,7 @@ export function setAngularAppTestingManifest(
   additionalServerAssets: Record<string, ServerAsset> = {},
   locale?: string,
   rootComponent: Type<unknown> = AppComponent,
+  extraProviders: Array<Provider | EnvironmentProviders> = [],
 ): void {
   destroyAngularServerApp();
 
@@ -89,6 +98,7 @@ export function setAngularAppTestingManifest(
           provideExperimentalZonelessChangeDetection(),
           provideRouter(routes),
           provideServerRoutesConfig(serverRoutes),
+          ...extraProviders,
         ],
       });
     },

